### ISSUE TITLE
How to Fix the Seed in in CTMs

### ISSUE THREAD

**Issue by Jobtimize** [MilaNLProc/contextualized-topic-models#56]
* Contextualized Topic Models version: __version__ = '1.8.2'
* Python version: 3.7.2
* Operating System: iOS locally and VM Linux Centos 7

First of all, thanks a lot for the awesome, nicely structured package! 

### Description

I'm trying to run the ZeroShot model over my personal train and test dataset, using the 'sbert.net_models_bert-base-nli-mean-tokens' BERT model. I succeed in training and evaluating the model, however, when I train two separate models with the same parameter settings, I get very different results (that is, very different topics)

I would like to get the same results for each combination of hyperparameter settings, so that I can actually see what the effect is of using different hyperparameters. 

### What I Did

```
def set_seed():
    """ Fixes the seed for every iteration.
        Should be placed in all torch loops. """
    torch.manual_seed(SETUP.SEED)
    torch.cuda.manual_seed(SETUP.SEED)
    np.random.seed(SETUP.SEED)
    random.seed(SETUP.SEED)
    torch.backends.cudnn.enabled = False
    torch.backends.cudnn.deterministic = True
```

I tried to fix the seed by calling the function above in each for loop related to PyTorch, however, this doesn't seem to solve the problem. 

If anyone can help me to create reproduceable results, please let me know!

**Comment by vinid** [comment]
Hello @Jobtimize,

I am looking into this, however, this kind of hard replication might be difficult to get (as you might have seen in the paper, what we generally do is that we run the topic model more times to get stable results).

I'll run some experiments

**Comment by Jobtimize** [comment]
Hi @vinid ,

I understand. Thanks a lot! I'm looking forward to your findings.

**Comment by Jobtimize** [comment]
Dear @vinid ,

Did you manage to run some experiments already? 
I look forward to hear from you.

Kind regards,

**Comment by vinid** [comment]
Still running some experiments to check which is the component that introduces randomization

**Comment by vinid** [comment]
Hello @Jobtimize 

You can take a look at this notebook here: https://colab.research.google.com/drive/10Z1g7stkKNqfszwCicZOL3QlOK--RuDE?usp=sharing

every time I re-run the training of the model, I get the same topics.

Note however that this might not be the best way to understand the effect of using different hyperparameters

**Comment by Jobtimize** [comment]
Hi @vinid ,

Thanks a lot for the quick help, this will definitely help me out!
